tpm/tpm_test.go

```diff
@@ -604,3 +606,33 @@ func TestForceClear(t *testing.T) {
 		t.Fatal("Couldn't clear the TPM without owner auth in physical presence mode:", err)
 	}
 }
+
+func TestNVWriteValue(t *testing.T) {
+	mathrand.Seed(time.Now().Unix())
+	rwc := openTPMOrSkip(t)
+	nvList, err := GetNVList(rwc)
+	if err != nil {
+		t.Fatalf("Couldn't open TPM connection: %v", err)
+	}
+	// If no indices are set, skip the test.
+	// Need DefineSpace-Function for implementing the other case
+	if len(nvList) == 0 {
+		t.Skip()
+	}
+	rndIndex := nvList[mathrand.Intn(len(nvList))]
+	rndIndexInfo, err := GetNVIndex(rwc, rndIndex)
+	if err != nil {
+		t.Fatalf("Couldn't retrieve index information: %v", err)
+	}
+
+	indexValue, err := NVReadValue(rwc, rndIndex, 0, rndIndexInfo.Size, nil)
+	if err != nil {
+		t.Fatalf("Couldn't read index value @ index: %v with error: %v", rndIndex, err)
+	}
+
+	err = NVWriteValue(rwc, rndIndex, 0, indexValue, nil)
+	if err != nil {
+		t.Fatalf("Couldn't write to index: %v with error: %v", rndIndex, err)
```

Comment:
Use `t.Errorf` if the test can continue or you are doing the final check in the test function.